powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/controller/api/RecoveryController.java

```diff
@@ -86,10 +95,13 @@ public ObjectResponse<ConfirmRecoveryCodeResponse> confirmRecoveryCode(@RequestB
      * @throws Exception In case the service throws exception.
      */
     @PostMapping("/lookup")
-    public ObjectResponse<LookupRecoveryCodesResponse> lookupRecoveryCodesRequest(@RequestBody ObjectRequest<LookupRecoveryCodesRequest> request) throws Exception {
-        logger.info("LookupRecoveryCodesRequest received: {}", request);
-        final ObjectResponse<LookupRecoveryCodesResponse> response = new ObjectResponse<>(service.lookupRecoveryCodes(request.getRequestObject()));
-        logger.info("LookupRecoveryCodesRequest succeeded: {}", response);
+    public ObjectResponse<LookupRecoveryCodesResponse> lookupRecoveryCodesRequest(@Valid @RequestBody ObjectRequest<LookupRecoveryCodesRequest> request) throws Exception {
+        final LookupRecoveryCodesRequest req = request.getRequestObject();
+        logger.info("action: lookupRecoveryCodesRequest, state: initiated, userId: {}, activationId: {}", req.getUserId(), req.getActivationId());
```

Comment:
Application ID could be also added.
```suggestion
        logger.info("action: lookupRecoveryCodesRequest, state: initiated, userId: {}, activationId: {}, applicationId: {}", req.getUserId(), req.getActivationId(), req.getApplicationId());
```